### ISSUE TITLE
RPG_RT move route edge case

### ISSUE THREAD

**Issue by fmatthew5876** [EasyRPG/Player#1588]
Create a common parallel event with this code:

```
Wait 0.0s
Move Player Up
Wait 0.0s
Move Player Right
Wait 0.0s
Loop:
  Wait 0.0s
```

* RPG_RT and Player: the player will move up and then right

Now change the code to this:
```
Move Player Up
Wait 0.0s
Move Player Right
Wait 0.0s
Loop:
  Wait 0.0s
```

* RPG_RT: player moves right
* Player: player moves up and then right

Now change the code to this:
```
Move Player Up
Wait 0.0s
Wait 0.0s
Move Player Right
Wait 0.0s
Loop:
  Wait 0.0s
```

* RPG_RT and Player: player moves up and right

Maybe an RPG_RT bug?

**Comment by Ghabry** [comment]
Wut. Don't tell me wine introduces other bugs

**Comment by fmatthew5876** [comment]
Parallel common event

```
Open Save Menu
Move Route: Player, Freq=8, Move Up
Open Save Menu
Move Route: Player, Freq=8, Move Right
Loop:
  OpenSaveMenu
```

| Frame | stop_count | max_stop_count | remaining_step | move_frequency | move_route_overwrite |
| --- | --- | --- | --- | --- | --- | 
| 1 | 0 | 0 | 0 | 2 | 1|
| 2 | 0 | 0 | 224 | 8 | 1 |
| 3 | 0 | 0 | 192 | 8 | 1 |
| 4 | 0 | 0 | 160 | 8 | 1 |
| 5 | 0 | 0 | 128 | 8 | 1 |
| 6 | 0 | 0 | 96 | 8 | 1 |
| 7 | 0 | 0 | 64 | 8 | 1 |
| 8 | 0 | 0 | 32 | 8 | 1 |
| 9 | 0 | 0 | 0 | 8 | 1 |
| 10 | 0 | 4 | 224 | 8 | 1 |
| 11 | 0 | 4 | 192 | 8 | 1 |
| 12 | 0 | 4 | 160 | 8 | 1 |
| 13 | 0 | 4 | 128 | 8 | 1 |
| 14 | 0 | 4 | 96 | 8 | 1 |
| 15 | 0 | 4 | 64 | 8 | 1 |
| 16 | 0 | 4 | 32 | 8 | 1 |
| 17 | 0 | 128 | 0 | 2 | 0 |
| 18 | 1 | 128 | 0 | 2 | 0 |

Parallel common event

```
Open Save Menu
Move Route: Player, Freq=7, Move Up
Open Save Menu
Move Route: Player, Freq=7, Move Right
Loop:
  OpenSaveMenu
```

| Frame | stop_count | max_stop_count | remaining_step | move_frequency | move_route_overwrite | move_route_index |
| --- | --- | --- | --- | --- | --- | 
| 1 | 0 | 0 | 0 | 2 | 1 | 0 | 
| 2 | 0 | 4 | 224 | 8 | 1 | 1 |
| 3 | 0 | 4 | 192 | 8 | 1 | 0 :question |
| 4 | 0 | 4 | 160 | 8 | 1 | 0 |
| 5 | 0 | 4 | 128 | 8 | 1 | 0 |
| 6 | 0 | 4 | 96 | 8 | 1 | 0 |
| 7 | 0 | 4 | 64 | 8 | 1 | 0 |
| 8 | 0 | 4 | 32 | 8 | 1 | 0 |
| 9 | 0 | 4 | 0 | 8 | 1 | 0 |
| 9 | 0 | 4 | 0 | 8 | 1 | 0 |
| 9 | 0 | 4 | 0 | 8 | 1 | 0 |
| 10 | 0 | 4 | 224 | 8 | 1 |
| 11 | 0 | 4 | 192 | 8 | 1 |
| 12 | 0 | 4 | 160 | 8 | 1 |
| 13 | 0 | 4 | 128 | 8 | 1 |
| 14 | 0 | 4 | 96 | 8 | 1 |
| 15 | 0 | 4 | 64 | 8 | 1 |
| 16 | 0 | 4 | 32 | 8 | 1 |
| 17 | 0 | 128 | 0 | 2 | 0 |
| 18 | 1 | 128 | 0 | 2 | 0 |

**Comment by fmatthew5876** [comment]
It looks like the difference is caused by this `stop_count = 0xFFFF` thing. In the case where we start with a move route, it looks like we wait 1 frame for stop count to go `65535` -> `0`. Since we wait 1 frame, the first move command never got to start when we overwrite it with the second one.

**Comment by fmatthew5876** [comment]
Move test cases

Parallel common event

```
Transfer Player: other map
Move Up
Wait 0.0s
Move Right
Loop:
  Wait 0.0s
```

* RPG_RT and Player: New game starts on other map, player moves right.

Parallel common event

```
Wait 0.0s
Transfer Player: other map
Move Up
Wait 0.0s
Move Right
Loop:
  Wait 0.0s
```

* RPG_RT and Player: new game loads on first map, player starts move slightly up by 1 frame, teleports to new map, player moves right

**Comment by fmatthew5876** [comment]
This is fixed in #1601

**Comment by fmatthew5876** [comment]
This is captured in #1567 test cases, so I will close this issue.
